oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,313 @@
+      " (expected ", limits.min_red, "..=", limits.max_red, "), G=", rgb[1],
+      " (expected ", limits.min_green, "..=", limits.max_green, "), B=", rgb[2],
+      " (expected ", limits.min_blue, "..=", limits.max_blue, ")\n"));
+}
+
+absl::Status CheckRgbaPixel(unsigned pixel, const ChannelLimits& limits,
+                            const sapi::v::Array<unsigned>& buffer) {
+  // RGBA images are upside down - adjust for normal ordering
+  unsigned adjusted_pixel = pixel % 128 + (127 - (pixel / 128)) * 128;
+
+  if (buffer.GetSize() <= adjusted_pixel) {
+    return absl::InternalError("Buffer overrun\n");
+  }
+
+  auto rgba = buffer[adjusted_pixel];
+  if (TIFFGetR(rgba) >= static_cast<unsigned>(limits.min_red) &&
+      TIFFGetR(rgba) <= static_cast<unsigned>(limits.max_red) &&
+      TIFFGetG(rgba) >= static_cast<unsigned>(limits.min_green) &&
+      TIFFGetG(rgba) <= static_cast<unsigned>(limits.max_green) &&
+      TIFFGetB(rgba) >= static_cast<unsigned>(limits.min_blue) &&
+      TIFFGetB(rgba) <= static_cast<unsigned>(limits.max_blue) &&
+      TIFFGetA(rgba) >= static_cast<unsigned>(limits.min_alpha) &&
+      TIFFGetA(rgba) <= static_cast<unsigned>(limits.max_alpha)) {
+    return absl::OkStatus();
+  }
+
+  return absl::InternalError(absl::StrCat(
+      "Pixel ", pixel, " did not match expected results.\n", "Got R=",
+      TIFFGetR(rgba), " (expected ", limits.min_red, "..=", limits.max_red,
+      "), G=", TIFFGetG(rgba), " (expected ", limits.min_green,
+      "..=", limits.max_green, "), B=", TIFFGetB(rgba), " (expected ",
+      limits.min_blue, "..=", limits.max_blue, "), A=", TIFFGetA(rgba),
+      " (expected ", limits.min_alpha, "..=", limits.max_alpha, ")\n"));
+}
+
+}  // namespace
+
+std::string GetFilePath(const std::string& dir, const std::string& filename) {
+  return sandbox2::file::JoinPath(dir, "test", "images", filename);
+}
+
+std::string GetFilePath(const std::string filename) {
+  std::string cwd = sandbox2::file_util::fileops::GetCWD();
+  auto find = cwd.rfind("build");
+
+  std::string project_path;
+  if (find == std::string::npos) {
+    LOG(ERROR)
+        << "Something went wrong: CWD don't contain build dir. "
+        << "Please run tests from build dir or send project dir as a "
+        << "parameter: ./sandboxed /absolute/path/to/project/dir .\n"
+        << "Falling back to using current working directory as root dir.\n";
+    project_path = cwd;
+  } else {
+    project_path = cwd.substr(0, find);
+  }
+
+  return sandbox2::file::JoinPath(project_path, "test", "images", filename);
+}
+
+absl::Status LibTIFFMain(const std::string& srcfile) {
+  // to use dir and file inside sapi-libtiff, use
+  // sandbox(file) – file only -- or
+  // sandbox(file, dir) -- file and dir -- or
+  // sandbox(nullopt, dir) -- dir only.
+  // file and directory must exist.
+  // all paths must be absolute.
+
+  TiffSapiSandbox sandbox(srcfile);
+
+  bool pixel_status = true;
```

Comment:
Maybe call it `pixel_status_ok` to make more clear what this boolean value means.